make/migrations/postgresql/0030_1.11.0_schema.up.sql

```diff
@@ -43,6 +43,19 @@ CREATE TABLE artifact_reference
   CONSTRAINT  unique_reference UNIQUE (parent_id, child_id)
 );
 
+/* artifact_trash records deleted artifact */
+CREATE TABLE artifact_trash
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  artifact_id   int NOT NULL,
+  media_type    varchar(255) NOT NULL,
+  manifest_media_type varchar(255) NOT NULL,
+  project_id    int NOT NULL,
+  repository_id int NOT NULL,
+  digest        varchar(255) NOT NULL,
+  creation_time timestamp default CURRENT_TIMESTAMP,
+  UNIQUE (artifact_id)
+);
```

Comment:
Unique artifact ID is meaningless as the artifact ID must be unique.
Unique(repo, digest)?